internal/ethapi/celo_api_test.go

```diff
@@ -531,3 +534,121 @@ func checkTxFields(
+// corresponds to Celo1 chain based on the provided chain configuration
+func Test_isCelo1Block(t *testing.T) {
+	cel2Time := uint64(1000)
+
+	t.Run("Non-Celo", func(t *testing.T) {
+		res := isCelo1Block(&params.ChainConfig{
+			Cel2Time: nil,
+		}, 1000)
+
+		assert.False(t, res)
+	})
+
+	t.Run("Celo1", func(t *testing.T) {
+		res := isCelo1Block(&params.ChainConfig{
+			Cel2Time: &cel2Time,
+		}, 500)
+
+		assert.True(t, res)
+	})
+
+	t.Run("Celo2", func(t *testing.T) {
+		res := isCelo1Block(&params.ChainConfig{
+			Cel2Time: &cel2Time,
+		}, 1000)
+
+		assert.False(t, res)
+	})
+}
+
+// TestRPCMarshalBlock_Celo1TotalDifficulty tests the RPCMarshalBlock function, specifically for totalDifficulty field
+// It validates the result has `totalDifficulty` field only if it's Celo1 block
+func TestRPCMarshalBlock_Celo1TotalDifficulty(t *testing.T) {
+	t.Parallel()
+
+	blockTime := uint64(1000)
+	block := types.NewBlock(&types.Header{Number: big.NewInt(100), Time: blockTime}, &types.Body{Transactions: []*types.Transaction{}}, nil, blocktest.NewHasher())
+
+	getExpectedJson := func(t *testing.T, expectedTd string) string {
+		data := map[string]interface{}{
+			"difficulty":       "0x0",
+			"extraData":        "0x",
+			"gasLimit":         "0x0",
+			"gasUsed":          "0x0",
+			"hash":             "0xd8371db7c48c7538a9cdf3e9211510e423039c0af3e1aa61d0a40960214d2101",
+			"logsBloom":        "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+			"miner":            "0x0000000000000000000000000000000000000000",
+			"mixHash":          "0x0000000000000000000000000000000000000000000000000000000000000000",
+			"nonce":            "0x0000000000000000",
+			"number":           "0x64",
+			"parentHash":       "0x0000000000000000000000000000000000000000000000000000000000000000",
+			"receiptsRoot":     "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
+			"sha3Uncles":       "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+			"size":             "0x1f7",
+			"stateRoot":        "0x0000000000000000000000000000000000000000000000000000000000000000",
+			"timestamp":        "0x3e8",
+			"transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
+			"uncles":           []interface{}{},
+		}
+
+		if expectedTd != "" {
+			data["totalDifficulty"] = expectedTd
+		}
+
+		marshaled, err := json.Marshal(&data)
+		require.NoError(t, err)
+
+		return string(marshaled)
+	}
```

Comment:
I don't think we need to be checking the whole marshalled json here. `RPCMarshalBlock` returns a map of string to interface which is encoded to json in a standard way by the go json package, so we don't need to re-test that. I think it is sufficient to call `RPCMarshalBlock` and simply check the value of the "totalDifficulty" key.

Reply:
Fixed 4502e6bfcec25ec024e6ca52db3fc754e56f2d05